### PR TITLE
Cleanup formatting tags

### PR DESCRIPTION
**Pull request by wheleph**

### CODE REVIEW

File: nttt/cleanup_html.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import re
-from .nttt_logging import display_tags
+from .nttt_logging import log_replacement
```
Comment (wheleph) [inline nttt/cleanup_html.py:2]: minor refactoring to unify logging in all cases

File: unit_test/test_cleanup_formatting.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        self.assertEqual(cleanup_formatting.trim_formatting_tags(init, self.logging), out)
+
+    def test_printer_friendly_link(self):
+        init = '[ printvriendelijke versie ]' \
+               '(https://projects.raspberrypi.org/en/projects/boat-race/print) {:target="_blank"}.'
+        out = '[ printvriendelijke versie ]' \
+              '(https://projects.raspberrypi.org/en/projects/boat-race/print){:target="_blank"}.'
+
+        self.assertEqual(cleanup_formatting.trim_formatting_tags(init, self.logging), out)
+
```
Comment (robertjank) [inline unit_test/test_cleanup_formatting.py:33]: Another situation I have seen is {:target="_ blank"}, i.e. a space between the underscore and the word blank. Maybe useful to add another test case, or include in test case test_fix_value.

File: nttt/cleanup_formatting.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import re
+from .nttt_logging import log_replacement
+
+
+def trim_formatting_tags(md_file_content, logging):
+    return re.sub(r'(?P<last_word>\S+?)\s*{\s*:\s*(?P<tag>[\w]+?)\s*=\s*"\s*(?P<value>.+?)\s*"\s*}',
```
Comment (robertjank) [inline nttt/cleanup_formatting.py:6]: See also the remark for test_cleanup_formatting.py. This regular expression doesn't fix the case {target="_ blank"}. This case occurs in the wild for example in project Boat race, language es-LA, file step_1.md.

**Review by robertjank — CHANGES_REQUESTED** [review]

**Comment by robertjank** [comment]
One more case that is not yet covered: project Boat race, language it-IT, file step_7.md contains
0\` {: Class = "block3variables"}
which is replaced with
0\`{:Class="block3variables"}
The word Class may have to be replaced with class. (Although it is probably converted to CSS, so it may not be needed to ensure the right formatting functionality.)

File: unit_test/test_cleanup_formatting.py (hunk 2 above)
Comment (wheleph) [reply unit_test/test_cleanup_formatting.py:33]: Good catch.
But how should we deal with those? Removing all spaces within quotes might be too aggressive.
I propose to fix only this specific situation, i. e. remove any spaces between "_" and "blank" within quotes.
What do you think about that?

**Comment by wheleph** [comment]
> One more case that is not yet covered: project Boat race, language it-IT, file step_7.md contains
> 0` {: Class = "block3variables"}
> which is replaced with
> 0`{:Class="block3variables"}
> The word Class may have to be replaced with class. (Although it is probably converted to CSS, so it may not be needed to ensure the right formatting functionality.)

@majamanojlovic is this really an issue? Shall we replace "Class" with "class" in the example that Robert-Jan demonstrated above. If we do then we need to come up with some specific rules (for example lowercase everything within `{: }` or only key names or only specific key names (like "class").

File: unit_test/test_cleanup_formatting.py (hunk 2 above)
Comment (robertjank) [reply unit_test/test_cleanup_formatting.py:33]: I fully agree with that. The risk that a translation engine or a human translator adds spaces in other string values (like Scratch class names) is probably very small and therefore doesn't have to be taken into account.

Comment (wheleph) [reply unit_test/test_cleanup_formatting.py:33]: done

**Review by robertjank — APPROVED** [review]

**Comment by majamanojlovic** [comment]
@wheleph @robertjank not sure if this is still an issue but yes - we want it to be in lowercase.

**Comment by robertjank** [comment]
Yes, @majamanojlovic, it is still an issue: see [#54](https://github.com/wheleph/nttt/issues/54).

**Pull request merged**